activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -695,7 +696,7 @@ def register_integer_type(mapping, key, options) # :nodoc:
       end
 
       def extract_precision(sql_type)
-        if /time/ === sql_type
+        if /\A(?:date)?time(?:stamp)?\b/ === sql_type
```

Comment:
`/time/ === sql_type` matches ENUM and SET types (e.g. `enum('date','time')`. related #22896, #23355).

Actually `extract_precision` is not called against these types, but I changed for consistency with this line.
https://github.com/rails/rails/pull/23553/files#diff-f50ea2a3acc95dd2652e9cb1b3f75961R44